lib/Ast_to_Mir.ml

```diff
@@ -44,11 +44,10 @@ let targetpe e =
   Assignment (t, BinOp (t, Plus, e))
 
 let trans_loc = function
-  | Ast.Nowhere -> ""
-  | Ast.Location (start, end_) ->
+  | {Ast.start_loc; end_loc} ->
       (* XXX hack hack *)
-      let open Lexing in
-      sprintf "\"%s\", line %d-%d" start.pos_fname start.pos_lnum end_.pos_lnum
+      sprintf "\"%s\", line %d-%d" start_loc.filename start_loc.linenum
```

Comment:
Yes, start and end can be in different files.

Generally, we have to make the decision of precisely what to print in errors. If we don't watch out it can be too much information, e.g. if we print the full include backtrace of both the start and end location. So are you proposing we do not print the end location at all in errors and just stick with the start location?